### PR TITLE
Allow decreasing delegate take

### PR DESCRIPTION
**Pull request by unconst**
Allows delegates the ability to decrease take on their delegates from 18 -> 0 % but only lower.

### CODE REVIEW

File: pallets/subtensor/src/lib.rs [hunk 1]
```diff
@@ -1279,6 +1281,38 @@ pub mod pallet {
             Self::do_become_delegate(origin, hotkey, Self::get_default_take())
         }
 
+        // --- Allows delegates to decrease its take value.
+        //
+        // # Args:
+        // 	* 'origin': (<T as frame_system::Config>Origin):
```
Comment (cuteolaf) [inline pallets/subtensor/src/lib.rs:1287]:
```suggestion
        // 	* 'origin': (<T as frame_system::Config>::Origin):
```

File: pallets/subtensor/src/staking.rs [hunk 2]
```diff
@@ -86,6 +86,87 @@ impl<T: Config> Pallet<T> {
         Ok(())
     }
 
+    // ---- The implementation for the extrinsic become_delegate: signals that this hotkey allows delegated stake.
```
Comment (cuteolaf) [inline pallets/subtensor/src/staking.rs]:
```suggestion
    // ---- The implementation for the extrinsic decrease_take
```

File: pallets/subtensor/src/staking.rs [hunk 3]
```diff
@@ -86,6 +86,87 @@ impl<T: Config> Pallet<T> {
...
+    //
+    // # Args:
+    // 	* 'origin': (<T as frame_system::Config>RuntimeOrigin):
```
Comment (cuteolaf) [inline pallets/subtensor/src/staking.rs:92]:
```suggestion
    // 	* 'origin': (<T as frame_system::Config>::RuntimeOrigin):
```

File: pallets/subtensor/src/lib.rs [hunk 4]
```diff
@@ -1279,6 +1281,38 @@ pub mod pallet {
...
+        // --- Allows delegates to decrease its take value.
+        //
+        // # Args:
+        // 	* 'origin': (<T as frame_system::Config>::Origin):
+        // 		- The signature of the caller's coldkey.
+        //
+        // 	* 'hotkey' (T::AccountId):
+        // 		- The hotkey we are delegating (must be owned by the coldkey.)
+        //
+        // 	* 'take' (u64):
```
Comment (cuteolaf) [inline pallets/subtensor/src/lib.rs]:
```suggestion
        // 	* 'take' (u16):
```

File: pallets/subtensor/src/lib.rs [hunk 5]
```diff
@@ -1279,6 +1281,38 @@ pub mod pallet {
...
+        // 		- The new stake proportion that this hotkey takes from delegations.
+        //
+        // # Event:
+        // 	* DelegateAdded;
+        // 		- On successfully setting a hotkey as a delegate.
```
Comment (cuteolaf) [inline pallets/subtensor/src/lib.rs]:
```suggestion
        // 	* TakeDecreased;
        // 		- On successfully setting a decreased take for this hotkey.
```

File: pallets/subtensor/src/staking.rs [hunk 6]
```diff
@@ -86,6 +86,87 @@ impl<T: Config> Pallet<T> {
...
+    //
+    // 	* 'hotkey' (T::AccountId):
+    // 		- The hotkey we are delegating (must be owned by the coldkey.)
+    //
+    // 	* 'take' (u16):
+    // 		- The stake proportion that this hotkey takes from delegations.
+    //
+    // # Event:
+    // 	* DelegateAdded;
+    // 		- On successfully setting a hotkey as a delegate.
```
Comment (cuteolaf) [inline pallets/subtensor/src/staking.rs]:
```suggestion
    // 	* TakeDecreased;
    // 		- On successfully setting a decreased take for this hotkey.
```

File: pallets/subtensor/src/staking.rs [hunk 7]
```diff
@@ -86,6 +86,87 @@ impl<T: Config> Pallet<T> {
...
+    //
+    // # Raises:
+    // 	* 'NotRegistered':
+    // 		- The hotkey we are delegating is not registered on the network.
+    //
+    // 	* 'NonAssociatedColdKey':
+    // 		- The hotkey we are delegating is not owned by the calling coldket.
+    //
+    // 	* 'TxRateLimitExceeded':
+    // 		- Thrown if key has hit transaction rate limit
```
Comment (cuteolaf) [inline pallets/subtensor/src/staking.rs:113]:
```suggestion
    // 	* 'InvalidTransaction':
    // 		- The delegate is setting a take which is not lower than the previous.
```

File: pallets/subtensor/src/staking.rs [hunk 8]
```diff
@@ -86,6 +86,87 @@ impl<T: Config> Pallet<T> {
...
+        // --- 1. We check the coldkey signature.
+        let coldkey = ensure_signed(origin)?;
+        log::info!(
+            "do_decrease_take( origin:{:?} hotkey:{:?}, take:{:?} )",
+            coldkey,
+            hotkey,
+            take
+        );
+
+        // --- 2. Ensure we are delegating an known key.
```
Comment (cuteolaf) [inline pallets/subtensor/src/staking.rs]:
```suggestion
        // --- 2. Ensure we are delegating a known key.
```

**Comment by distributedstatemachine** [comment]
Closing re: https://github.com/opentensor/subtensor/pull/313#issuecomment-2050711229 . A new PR will be opened

**Pull request closed without merging**